ui/src/app/applications/components/application-resource-tree/application-resource-tree.tsx

```diff
@@ -407,21 +419,34 @@ function renderPodGroup(props: ApplicationResourceTreeProps, id: string, node: R
     const margin = 8;
     let topExtra = 0;
     const podGroup = node.podGroup;
+    const podGroupHealthy = podGroup.pods.filter(pod => pod.health === 'Healthy');
+    const podGroupDegraded = podGroup.pods.filter(pod => pod.health === 'Degraded');
+    const podGroupInProgress = podGroup.pods.filter(pod => pod.health === 'Progressing');
+
+    const showPodGroupByStatus = props.tree.nodes.filter((rNode: ResourceTreeNode) => rNode.kind === 'Pod').length >= props.podGroupCount;
+    const numberOfRows = showPodGroupByStatus
+        ? [podGroupHealthy, podGroupDegraded, podGroupInProgress].reduce((total, podGroupByStatus) => total + (podGroupByStatus.filter(pod => pod).length > 0 ? 1 : 0), 0)
+        : Math.ceil(podGroup.pods.length / 8);
+
     if (podGroup) {
-        const numberOfRows = Math.ceil(podGroup.pods.length / 8);
         topExtra = margin + (POD_NODE_HEIGHT / 2 + 30 * numberOfRows) / 2;
     }
```

Comment:
This is the place where I determine how many rows are needed.  Just a comment for now, when you fix the vertical spacing in a follow-on PR.

Reply:
I improved the alignment slightly but will try to adjust further in the follow-up PR.

Reply:
Ok